src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -312,16 +311,7 @@ protected void onStartup() {
     private void validateSequenceDictionaries() {
         final SAMSequenceDictionary refDict = hasReference() ? reference.getSequenceDictionary() : null;
         final SAMSequenceDictionary readDict = hasReads() ? reads.getSequenceDictionary() : null;
-        List<SAMSequenceDictionary> variantDicts = new ArrayList<>();
-        if (hasFeatures()){
-            List<VCFHeader> variantHeaders = features.getAllVariantHeaders();
-            for (VCFHeader header : variantHeaders) {
-                SAMSequenceDictionary headerDict = header.getSequenceDictionary();
-                if (headerDict != null) {
-                    variantDicts.add(headerDict);
-                }
-            }
-        }
+        final List<SAMSequenceDictionary> variantDicts = hasFeatures() ? features.getAllSequenceDictionaries() : Collections.emptyList();
```

Comment:
Also, it is problematic that this method is looking specifically for "variant" dictionaries, but you've now modified it to potentially get dictionaries from other kinds of Feature files (via the indices). This either needs to be clearly documented, or you need to return to the previous behavior of only using variant (VCF/BCF) dictionaries during validation.